### PR TITLE
Adding a way to run an Athens in docker

### LINKED ISSUE #420: Fully Dockerized Development Environment
Sorry if this is a duplicate issue. But it would be very helpful to onboard new contributors and anyone who just wants to interact with Athens with nothing but docker (no buffalo needed, not even Go is needed) 

Let's have our dev set up to two tiers: 

1. Just spin things up, I hate installing things. 
`docker-compose up athens-dev` -- this will spin up the minimum dependencies but as well as two docker images (one for proxy and one for athens) that will already have buffalo and a module-enabled-go binary in them. So all you have to do is ping localhost:3000 

2. I'd like to contribute and know how things work and be able to develop fast. 
We already have this, just good old manual install of Go, Buffalo, and >go1.11beta3

### PR DESCRIPTION
**Pull request by arschles**
**What is the problem I am trying to address?**

There's no easy way to start up Athens along with its service dependencies completely in Docker

**How is the fix applied?**

I added a docker-compose target and a Makefile target for starting athens + dependencies completely inside Docker, and I added instructions in the development.md file to use it all

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes #420

cc/ @marwan-at-work. Is this what you were going for in #420?

### CODE REVIEW

**Comment by arschles** [comment]
This partially replaces #888. That PR was really old and was taking longer to resolve conflicts than it was to open a new PR

**Comment by chriscoffee** [comment]
LGTM :shipit:

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -17,13 +17,68 @@ GO_BINARY_PATH=go1.11.X
...
+
+1. Builds Athens from scratch
+2. Starts up [MongoDB](https://www.mongodb.com/) and [Jaeger](https://www.jaegertracing.io/)
+3. Configures Athens to use MongoDB for its storage and Jaeger for its distributed tracing
+4. Runs Athens
+
+You'll need [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/) installed. Once you do, run the below command to set everything up:
+
+```console
+$ make dev-docker
```
Comment (chriscoffee) [inline DEVELOPMENT.md]: Just a small typo here, should be `make run-docker`. I think it used to be `dev-docker` right? Looks like the teardown is right though.

Comment (arschles) [reply DEVELOPMENT.md]: good catch, thanks @chriscoffee! Change coming

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -17,13 +17,68 @@ GO_BINARY_PATH=go1.11.X
...
+```console
+docker-compose -p athensdockerdev up -d dev
+Creating network "athensdockerdev_default" with the default driver
+Creating athensdockerdev_jaeger_1 ... done
+Creating athensdockerdev_minio_1  ... done
+Creating athensdockerdev_mongo_1  ... done
+Creating athensdockerdev_dev_1    ... done
+```
+
+After that runs, Athens should be up and running and serving on post 3000. To test it out, run this command:
```
Comment (michalpristas) [inline DEVELOPMENT.md]: s/post/port

Comment (arschles) [reply DEVELOPMENT.md]: fixed, thanks @michalpristas !

**Comment by arschles** [comment]
@marwan-at-work I see what you're saying. The documentation that I changed is in `DEVELOPMENT.md`, so I feel like it assumed that the reader has already checked the code out. What do you think?

**Pull request merged**